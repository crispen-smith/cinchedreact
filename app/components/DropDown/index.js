/**
 *
 * DropDown
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import styled from 'styled-components';

/* eslint-disable react/prefer-stateless-function */
class DropDown extends React.PureComponent {
  render() {
    const options = this.props.options.map(option => (
      <option key={option} value={option}>
        {option}
      </option>
    ));

    return (
      <div className="form-row-primary">
        <label htmlFor={this.props.inputName}>
          {this.props.inputLabel}
          <select id={this.props.inputName} onChange={this.props.onChange}>
            {options}
          </select>
        </label>
      </div>
    );
  }
}

DropDown.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  inputLabel: PropTypes.string.isRequired,
  inputName: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default DropDown;
