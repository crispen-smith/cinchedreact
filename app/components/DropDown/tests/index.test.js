import React from 'react';
import renderer from 'react-test-renderer';

import DropDown from '../index';

describe('<DropDown />', () => {
  const InputName = 'TEST_NAME';
  const InputLabel = 'TEST_LABEL';
  const onChange = () => 1;
  const Options = ['First', 'Second', 'Third'];

  it('Renders a NameBox', () => {
    const tree = renderer
      .create(
        <DropDown
          inputLabel={InputLabel}
          inputName={InputName}
          onChange={onChange}
          options={Options}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
