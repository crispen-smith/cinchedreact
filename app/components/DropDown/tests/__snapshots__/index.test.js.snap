// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDown /> Renders a NameBox 1`] = `
<div
  className="form-row-primary"
>
  <label
    htmlFor="TEST_NAME"
  >
    TEST_LABEL
    <select
      id="TEST_NAME"
      onChange={[Function]}
    >
      <option
        value="First"
      >
        First
      </option>
      <option
        value="Second"
      >
        Second
      </option>
      <option
        value="Third"
      >
        Third
      </option>
    </select>
  </label>
</div>
`;
